src/Core/IToolWindowService.cs

```diff
@@ -43,5 +43,7 @@ public interface IToolWindowService
         /// </para>
         /// </remarks>
         void EnsureToolWindowExists(Guid toolWindowId);
+
+        T GetToolWindow<T>() where T : class;
```

Comment:
FYI it's possible to add specific type or interface to the `where` clause e.g. 
```
T GetToolWindow<T>() where T: class, IToolWindowService
```

However, we can't do that here unless we add a reference to the VS assemblies, which is what we were trying to avoid.